x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFunction.java

```diff
@@ -33,12 +36,17 @@ protected Object doFold(ZonedDateTime dateTime) {
         return dateTimeChrono(dateTime, extractor.chronoField());
     }
 
+    public static Integer dateTimeChrono(OffsetTime time, String tzId, String chronoName) {
+        return dateTimeChrono(time.withOffsetSameInstant(ZoneId.of(tzId).getRules().getOffset(Instant.now())),
```

Comment:
Since a `ZoneId` can be `ZoneOffset`, the conversion process should be promoted to a static method in `DateUtils`.